src/app/page.tsx

```
@@ -386,54 +437,130 @@ function mapDerivStatusToLocal(derivStatus?: DerivContractStatusData['status']):
+    currentInstrument,
+    globalOfferingsData,  // Primary new dependency
+    isLoadingGlobalOfferings,
+    globalOfferingsError,
+    userInfo, // Still needed for early exit
+    toast,
+    tradeDuration, // To check if current selection is still valid
+    setTradeDuration,
+    setAvailableDurations,
+    setIsLoadingDurations,
+    setIsTradeable,
+    logAutomatedTradingEvent // For logging issues
```

Biome (1.9.4)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)